### PR TITLE
Create a PR for issue-840

### PR DESCRIPTION
**Pull request by deepakapj**
Fix for Hystrix dashboard doesn't load Metrics DOTNET 6.0

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/SteeltoeOSS/Steeltoe?pullRequest=847) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/SteeltoeOSS/Steeltoe?pullRequest=847) before we can accept your contribution.<br/><br/>:x: deepakapj [sign now](https://cla.dotnetfoundation.org/SteeltoeOSS/Steeltoe?pullRequest=847)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/SteeltoeOSS/Steeltoe?pullRequest=847) it.</sub>

**Comment by dtillman** [comment]
@deepakapj : Thanks .. LGTM... can you please sign CLA .. thanks again!

File: src/CircuitBreaker/src/HystrixBase/Serial/SerialHystrixDashboardData.cs [hunk 1]
```diff
@@ -29,23 +29,45 @@ public static string ToJsonString(HystrixCommandMetrics commandMetrics)
...
             using var sw = new StringWriter();
             using var writer = new JsonTextWriter(sw);
-            WriteCommandMetrics(writer, commandMetrics);
+            try
+            {
+                WriteCommandMetrics(writer, commandMetrics);
+            }
+            catch (Exception)
+            { 
+
```
Comment (TimHess) [inline src/CircuitBreaker/src/HystrixBase/Serial/SerialHystrixDashboardData.cs:38]:
```suggestion
```

File: src/CircuitBreaker/src/HystrixBase/Serial/SerialHystrixDashboardData.cs [hunk 2]
```diff
@@ -29,23 +29,45 @@ public static string ToJsonString(HystrixCommandMetrics commandMetrics)
...
             using var writer = new JsonTextWriter(sw);
-            WriteThreadPoolMetrics(writer, threadPoolMetrics);
+            try
+            {
+                WriteThreadPoolMetrics(writer, threadPoolMetrics);
+  
+            }
+            catch (Exception)
+            {
+
```
Comment (TimHess) [inline src/CircuitBreaker/src/HystrixBase/Serial/SerialHystrixDashboardData.cs:54]:
```suggestion
```

File: src/CircuitBreaker/src/HystrixBase/Serial/SerialHystrixDashboardData.cs [hunk 3]
```diff
@@ -29,23 +29,45 @@ public static string ToJsonString(HystrixCommandMetrics commandMetrics)
...
 
         public static string ToJsonString(HystrixThreadPoolMetrics threadPoolMetrics)
         {
             using var sw = new StringWriter();
             using var writer = new JsonTextWriter(sw);
-            WriteThreadPoolMetrics(writer, threadPoolMetrics);
+            try
+            {
+                WriteThreadPoolMetrics(writer, threadPoolMetrics);
+
```
Comment (TimHess) [inline src/CircuitBreaker/src/HystrixBase/Serial/SerialHystrixDashboardData.cs:50]:
```suggestion
```

File: src/CircuitBreaker/src/HystrixBase/Serial/SerialHystrixDashboardData.cs [hunk 4]
```diff
@@ -29,23 +29,45 @@ public static string ToJsonString(HystrixCommandMetrics commandMetrics)
...
             using var sw = new StringWriter();
             using var writer = new JsonTextWriter(sw);
-            WriteCollapserMetrics(writer, collapserMetrics);
+            try
+            {
+                WriteCollapserMetrics(writer, collapserMetrics);
+            }
+            catch (Exception)
+            {
+
```
Comment (TimHess) [inline src/CircuitBreaker/src/HystrixBase/Serial/SerialHystrixDashboardData.cs:69]:
```suggestion
```

File: src/CircuitBreaker/src/HystrixBase/Serial/SerialHystrixDashboardData.cs [hunk 5]
```diff
@@ -29,23 +29,41 @@ public static string ToJsonString(HystrixCommandMetrics commandMetrics)
...
             using var sw = new StringWriter();
             using var writer = new JsonTextWriter(sw);
-            WriteCommandMetrics(writer, commandMetrics);
+            try
+            {
+                WriteCommandMetrics(writer, commandMetrics);
+            }
+            catch (Exception)
+            { 
+            }
```
Comment (TimHess) [inline src/CircuitBreaker/src/HystrixBase/Serial/SerialHystrixDashboardData.cs:38]:
```suggestion
            }

```

File: src/CircuitBreaker/src/HystrixBase/Serial/SerialHystrixDashboardData.cs [hunk 6]
```diff
@@ -29,23 +29,41 @@ public static string ToJsonString(HystrixCommandMetrics commandMetrics)
...
             using var sw = new StringWriter();
             using var writer = new JsonTextWriter(sw);
-            WriteThreadPoolMetrics(writer, threadPoolMetrics);
+            try
+            {
+                WriteThreadPoolMetrics(writer, threadPoolMetrics);
+            }
+            catch (Exception)
+            {
+            }
```
Comment (TimHess) [inline src/CircuitBreaker/src/HystrixBase/Serial/SerialHystrixDashboardData.cs:52]:
```suggestion
            }

```

File: src/CircuitBreaker/src/HystrixBase/Serial/SerialHystrixDashboardData.cs [hunk 7]
```diff
@@ -29,23 +29,41 @@ public static string ToJsonString(HystrixCommandMetrics commandMetrics)
...
             using var sw = new StringWriter();
             using var writer = new JsonTextWriter(sw);
-            WriteCollapserMetrics(writer, collapserMetrics);
+            try
+            {
+                WriteCollapserMetrics(writer, collapserMetrics);
+            }
+            catch (Exception)
+            {
+            }
```
Comment (TimHess) [inline src/CircuitBreaker/src/HystrixBase/Serial/SerialHystrixDashboardData.cs:66]:
```suggestion
            }

```

File: src/CircuitBreaker/src/HystrixBase/Serial/SerialHystrixDashboardData.cs [hunk 8]
```diff
@@ -29,23 +29,44 @@ public static string ToJsonString(HystrixCommandMetrics commandMetrics)
         {
             using var sw = new StringWriter();
             using var writer = new JsonTextWriter(sw);
-            WriteCommandMetrics(writer, commandMetrics);
+            try
+            {
+                WriteCommandMetrics(writer, commandMetrics);
+            }
+            catch (Exception)
+            {
```
Comment (TimHess) [inline src/CircuitBreaker/src/HystrixBase/Serial/SerialHystrixDashboardData.cs:37]:
```suggestion
            {
                // failing to write metrics should not crash the app
```

File: src/CircuitBreaker/src/HystrixBase/Serial/SerialHystrixDashboardData.cs [hunk 9]
```diff
@@ -29,23 +29,44 @@ public static string ToJsonString(HystrixCommandMetrics commandMetrics)
...
         {
             using var sw = new StringWriter();
             using var writer = new JsonTextWriter(sw);
-            WriteThreadPoolMetrics(writer, threadPoolMetrics);
+            try
+            {
+                WriteThreadPoolMetrics(writer, threadPoolMetrics);
+            }
+            catch (Exception)
+            {
```
Comment (TimHess) [inline src/CircuitBreaker/src/HystrixBase/Serial/SerialHystrixDashboardData.cs:52]:
```suggestion
            {
                // failing to write metrics should not crash the app
```

File: src/CircuitBreaker/src/HystrixBase/Serial/SerialHystrixDashboardData.cs [hunk 10]
```diff
@@ -29,23 +29,44 @@ public static string ToJsonString(HystrixCommandMetrics commandMetrics)
...
         {
             using var sw = new StringWriter();
             using var writer = new JsonTextWriter(sw);
-            WriteCollapserMetrics(writer, collapserMetrics);
+            try
+            {
+                WriteCollapserMetrics(writer, collapserMetrics);
+            }
+            catch (Exception)
+            {
```
Comment (TimHess) [inline src/CircuitBreaker/src/HystrixBase/Serial/SerialHystrixDashboardData.cs:67]:
```suggestion
            {
                // failing to write metrics should not crash the app
```

**Review by TimHess — COMMENTED** [review]
add comments to clear warnings

**Review by dtillman — APPROVED** [review]

**Comment by deepakapj** [comment]
Hi Tim & Team,

Any item pending from my side. Please let me know.

Thanks
Deepak Saxena

**Comment by TimHess** [comment]
All set here, thanks @deepakapj

**Comment by deepakapj** [comment]
Hi @TimHess & @dtillman ,

Could you please add this for 3.2 pre-release or release so that we can use in our project? This is blocker to release our project.
if this is part 3.2 pre-release then we can start development.

Thanks,

**Pull request merged**